WallsAndHoles/mapcell.cpp

```diff
@@ -50,3 +53,16 @@ void MapCell::tileChanged()
     else
         mGraphics->setBrush(Qt::NoBrush);
 }
+
+void MapCell::changeHeightMap(int x, int y, QColor color){
```

Comment:
I don't like the strategy of recreating the item every time. Really you should just treat it like the others, and change the color to clear when not in use. Really really this should just change mGraphics, as it is covered by mHeightMap anyways.

Also, when you paint on the height map, mHeightMap is not updated until you close, and reopen it which is not ideal.